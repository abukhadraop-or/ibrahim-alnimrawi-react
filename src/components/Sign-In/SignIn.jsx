import { FormInput, SubmitButton } from 'components/shared/Input/input.style';
import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';

/**
 * Render a <SignIn> component
 *
 * @return {JSX.Element}
 */
const SignIn = () => (
  <BrowserRouter>
    <h1>Sign In</h1>
    <NavLink to="/SignUp">Need an account? </NavLink>
    <FormInput type="email" placeHolder="Email" name="email" id="email" />
    <FormInput
      type="password"
      placeHolder="Password"
      name="password"
      id="password"
    />
    <SubmitButton>Sign in</SubmitButton>
  </BrowserRouter>
);

export default SignIn;
