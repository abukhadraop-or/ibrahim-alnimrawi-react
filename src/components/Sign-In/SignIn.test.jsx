import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from './SignIn';

Enzyme.configure({ adapter: new Adapter() });

describe('<SignIn />', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<SignIn />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
