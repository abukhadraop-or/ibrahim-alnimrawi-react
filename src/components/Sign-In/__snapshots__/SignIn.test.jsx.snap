// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should match the snapshot 1`] = `
Array [
  <h1>
    Sign In
  </h1>,
  <a
    aria-current={null}
    href="/SignUp"
    onClick={[Function]}
  >
    Need an account? 
  </a>,
  <input
    id="email"
    name="email"
    placeholder="Email"
    type="email"
  />,
  <input
    id="password"
    name="password"
    placeholder="Password"
    type="password"
  />,
  .c0 {
  background-color: #5cb85c;
  border: none;
  color: #ffffff;
  width: 7rem;
  height: 3rem;
  border-radius: 0.3rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  font-size: 1.25rem;
  padding: 0.75rem 1.5rem;
}

<button
    className="c0"
  >
    Sign in
  </button>,
]
`;

exports[`<SignIn /> should match the snapshot be 1`] = `
Array [
  <h1>
    Sign In
  </h1>,
  <a
    aria-current={null}
    href="/SignUp"
    onClick={[Function]}
  >
    Need an account? 
  </a>,
  <input
    id="email"
    name="email"
    placeholder="Email"
    type="email"
  />,
  <input
    id="password"
    name="password"
    placeholder="Password"
    type="password"
  />,
  .c0 {
  background-color: #5cb85c;
  border: none;
  color: #ffffff;
  width: 7rem;
  height: 3rem;
  border-radius: 0.3rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  font-size: 1.25rem;
  padding: 0.75rem 1.5rem;
}

<button
    className="c0"
  >
    Sign in
  </button>,
]
`;
